### PR TITLE
Changes to build and run Titan modules and applications with stock Lua (without changes needed to `luaconf.h`)

### PR DESCRIPTION
**Pull request by mascarenhas**
Link Titan modules statically with part of the Lua object files (only change over a regular Lua installation is that they need to be compiled with `-fPIC`, but no changes to `luaconf.h` are needed.

All `static` functions that every Titan module/app uses should be in a separate `titan.c` file (some of these functions may be compiled from Lua and specialized, with proper attribution).

Titan applications do not need Lua object files compiled with `-fPIC`, for now we can just have one set of object files, in the future we might want to have two.

I am also trying to get Travis to do OSX builds of Titan, but as I do not have a local OSX machine myself I will leave this out of this PR if it turns out to be too much trouble.

### CODE REVIEW

**Comment by hishamhm** [comment]
> I am also trying to get Travis to do OSX builds of Titan, but as I do not have a local OSX machine myself I will leave this out of this PR if it turns out to be too much trouble.

I'm going to push a commit here attempting to fix this, I checked the changes against the LuaRocks `.travis.yml`.

**Comment by mascarenhas** [comment]
Right now it seems like the Travis OSX worker is busted, it is erroring while downloading Python.

**Comment by mascarenhas** [comment]
Something pretty odd just happened with this branch, it is showing the latest commits that were already merged into `master`.

**Comment by hishamhm** [comment]
@mascarenhas my bad, just refresh :) I messed it up for a second when force-pushing a fixed-fix.

**Comment by mascarenhas** [comment]
We can already remove the `lua` folder from the repository, by the way. About this PR, I still have not tested if Titan executables are building correctly (I suspect not), and after fixing this we can merge and do the specialization of `getgeneric` some other time.

**Comment by mascarenhas** [comment]
Looks like the FFI needs some fixes to work on OSX. We should just leave the Travis build commented out for now and try to do these fixes in  another PR, preferably with access to an OSX machine.

**Comment by hishamhm** [comment]
All right, macOS ran, but c-parser is failing to parse some standard library headers there (those are usually the ones that push the parser the hardest).

**Comment by hishamhm** [comment]
@mascarenhas agreed!

File: titan-runtime/titan.h
Comment (hishamhm) [inline titan-runtime/titan.h]: getgeneric is duplicated here

Comment (mascarenhas) [reply titan-runtime/titan.h]: oops, pushing a fix

**Comment by mascarenhas** [comment]
Ok, will make sure we can still compile Titan applications, and not just Lua modules, later.

**Comment by mascarenhas** [comment]
All systems go @hishamhm!

**Review by hishamhm — APPROVED** [review]
@mascarenhas last missing bit: could you update the build/install instructions in README.md? I'll leave the PR approved so you can merge once that is in!

**Pull request merged**
